specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.benefitrecommendations.json

```diff
@@ -0,0 +1,486 @@
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/benefitRecommendationsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{billingScope}/providers/Microsoft.CostManagement/benefitRecommendations/{recommendationId}": {
+      "get": {
+        "tags": [
+          "BenefitRecommendations"
+        ],
+        "description": "Details of a benefit plan recommendation for what-if analysis.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/CostManagement/"
+        },
+        "x-ms-examples": {
+          "BenefitRecommendationById": {
+            "$ref": "./examples/BenefitRecommendationsByRecommendationId.json"
+          }
+        },
+        "operationId": "BenefitRecommendation_Get",
+        "parameters": [
+          {
+            "$ref": "common-types.json#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingScopeBenefitRecommendationParameter"
+          },
+          {
+            "name": "recommendationId",
+            "in": "path",
+            "description": "The ID of the recommendation.",
+            "required": true,
+            "type": "string",
+            "x-ms-parameter-location": "method",
+            "x-ms-skip-url-encoding": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/benefitRecommendationModel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "benefitRecommendationsListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The list of benefit recommendations.",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/benefitRecommendationModel"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The link (url) to the next page of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Result of listing benefit recommendations."
+    },
+    "benefitRecommendationModel": {
+      "type": "object",
+      "description": "benefit plan recommendation details.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```

Comment:
>Resource [](http://example.com/codeflow?start=33&length=8)

Inherit ProxyResoruce or TrackedResource

Reply:
Can't inherit from common types v2/v3
Our object has four properties id, name, type, kind
In common types v2, kind property is not present
In common types v3, kind property is not present and systemData is present which we don't need

Reply:
You can add 'kind' property on your own models, as well as inheriting the common types. Because 'kind' is not present on all resource types, it is not present on the common types.

Reply:
Got it. Used common types v2